collect/collect.go

```diff
@@ -395,16 +395,25 @@ func (i *InMemCollector) dealWithSentTrace(keep bool, sampleRate uint, sp *types
 	}
 	if keep {
 		i.Logger.Debug().WithField("trace_id", sp.TraceID).Logf("Sending span because of previous decision to send trace")
-		if sp.SampleRate < 1 {
-			sp.SampleRate = 1
-		}
-		sp.SampleRate *= sampleRate
+		mergeTraceAndSpanSampleRates(sp, sampleRate)
 		i.Transmission.EnqueueSpan(sp)
 		return
 	}
 	i.Logger.Debug().WithField("trace_id", sp.TraceID).Logf("Dropping span because of previous decision to drop trace")
 }
 
+func mergeTraceAndSpanSampleRates(sp *types.Span, traceSampleRate uint) {
+	if traceSampleRate != 1 {
```

Comment:
Missed adding it initial review, this is where I think we should default sample rate if negative or 0.

```suggestion
	if sp.SampleRate < 1 {
		sp.SampleRate = 1
	}
	sp.SampleRate *= sampleRate	if traceSampleRate != 1 {
	if traceSampleRate != 1 {
```

Reply:
I updated `mergeTraceAndSpanSampleRates` to set to 1 when something less than 1 and added a unit test to capture that behavior. I very briefly tried to find mention of the range of allowed values for SampleRate in documentation for OpenTelemetry but came up empty. Is there a source for that requirement so we can leave a breadcrumb of why?

Reply:
Honeycomb's use and interpretation of SampleRate predates OTel. OTel uses a p-value, which is effectively the mathematical inverse of Honeycomb's SampleRate. 

Honeycomb's backend data system interprets a missing value for SampleRate as a value of 0, so therefore converts those to 1 automatically so the math works correclty. We have been trying to push that idea upstream where we can; it benefits everyone to have it more explicit.